Fix: disallow non-super users to call has_schema_privilege (#13373)
crate/crate: docs/appendices/release-notes/unreleased.rst

.. stable branch. You can add a version label (`v/X.Y`) to the pull request for
.. an automated mergify backport.

> - Fixed an issue that allowed non-super users to check other users' privileges
>   by calling :ref:`has_schema_privilege <scalar-has-schema-priv>` function.

Reviewer: ```suggestion
- Fixed an issue that allowed users without the related privileges to check 
  other users' privileges by calling 
  :ref:`has_schema_privilege <scalar-has-schema-priv>` function.
```
